### PR TITLE
test: fixed connection lost test

### PR DESCRIPTION
**Pull request by jshrt**

### CODE REVIEW

File: test/util/timeout-registrySpec.ts [hunk 1]
```diff
@@ -121,7 +126,10 @@ describe('timeout registry', () => {
 
         it('clears timeout when connection lost', done => {
```
Comment (MuriloFrade) [inline test/util/timeout-registrySpec.ts]: Does this test do the same thing that the test above? They have the same description.

File: test/util/timeout-registrySpec.ts [hunk 2]
```diff
@@ -121,7 +126,10 @@ describe('timeout registry', () => {
...
             services.simulateConnectionLost()
-            setTimeout(done, 10)
+            setTimeout(() => {
+                sinon.assert.callCount(spy, 0)
+                done()
+            }, 50)
```
Comment (MuriloFrade) [inline test/util/timeout-registrySpec.ts]: What 50 means here ? 

Does it work like that also? 
```
services.simulateConnectionLost()
sinon.assert.notCalled(spy)
timeout(done, delay)
```

File: test/util/timeout-registrySpec.ts [hunk 3]
```diff
@@ -70,6 +71,10 @@ describe('timeout registry', () => {
         })
 
         it('clears timeout when connection lost', done => {
+            services.loggerMock
+                .expects('warn')
+                .never()
+
```
Comment (MuriloFrade) [inline test/util/timeout-registrySpec.ts]: You still need to test if the timeoutRegistry has called the timeRegistry.remove service for each timeout, no?

File: test/connection/connectionSpec.ts [hunk 4]
```diff
@@ -289,6 +289,7 @@ describe('connection', () => {
     await sendChallengeResponse()
     await receiveChallengeAccept()
     await receiveConnectionError()
+    await BBPromise.delay(10)
```
Comment (MuriloFrade) [inline test/connection/connectionSpec.ts]: you could use some variable to store the 10 value (`reconnectIntervalIncrement` for example) and use this in more places. This should make it easy to understand what '10'  means in this test.

File: test/util/timeout-registrySpec.ts (hunk 3 above)
Comment (jshrt) [reply test/util/timeout-registrySpec.ts]: I agree, it should actually assert for that, however for now `onConnectionLost` clears timeouts using `clearTimeout` without relying on the `timerRegistry` service. I wonder whether `onConnectionLost` should be refactored to use the services instead?

**Comment by yasserf** [comment]
status of PR?

**Pull request closed without merging**